Ginger/GingerCoreCommon/VariablesLib/VariableSelectionList.cs

```diff
@@ -80,12 +84,73 @@ public override bool SetValue(string value)
                 Value = value;
                 return true;
             }
+            else if (IsDynamicValueModificationEnabled)
+            {
+                OptionalValuesList.Add(new OptionalValue(value));
+                Value = OptionalValuesList[OptionalValuesList.Count - 1].Value;
+                return true;
+            }
             else
             {
                 return false;
             }
         }
 
+        public void DynamicDeleteValue(string value, ref string errorMsg)
+        {
+            if (IsDynamicValueModificationEnabled)
+            {
+                OptionalValue op;
+                if (OptionalValuesList.Where(pv => pv.Value == value).FirstOrDefault() != null)
+                {
+                    op = OptionalValuesList.Where(pv => pv.Value == value).FirstOrDefault();
+                }
+                else
+                {
+                    errorMsg = "Could not found the value entered for deletion";
+                    return;
+                }
+                int index = OptionalValuesList.IndexOf(op);
```

Comment:
no need "index", we can just check if list is empty after delete and then set emty value to 'Value' and check if deleted value was lat selected value and in this case if list is not empty then to select the List[0] value (first one)